src/App.tsx

```
@@ -1,37 +1,60 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
+
+type State = {
+  clockName: string;
+  hasClock: boolean;
+};
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends React.Component<{}, State> {
+  state: State = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerId?: number;
+
+  componentDidMount(): void {
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentWillUnmount(): void {
+    window.clearInterval(this.timerId);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  render() {
+    // const today = new Date();
+    // let clockName = 'Clock-0';
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    // This code starts a timer
 
-      <div className="Clock">
-        <strong className="Clock__name">{clockName}</strong>
+    // this code stops the timer
 
-        {' time is '}
+    const { hasClock, clockName } = this.state;
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+    document.addEventListener('contextmenu', (event: MouseEvent) => {
+      event.preventDefault();
+      this.setState({ hasClock: false });
+    });
```

It's good that you're preventing the default context menu behavior and updating the state to hide the clock on a right click. However, according to checklist item 1, you should remove this event listener in the `componentWillUnmount` method to avoid potential memory leaks. Consider moving this logic to a separate method and then removing the event listener in `componentWillUnmount`.